content/core-features/presence-and-occupancy/topic-presence_ia-update.textile

```diff
@@ -0,0 +1,737 @@
+await channel.Presence.EnterClientAsync("Mary", null); /* => Mary entered realtime-chat */
+```
+
+```[objc]
+ARTRest* rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+// request a wildcard token
+ARTTokenParams *tokenParams = [[ARTTokenParams alloc] initWithClientId:@"*"];
+[rest.auth requestToken:tokenParams withOptions:nil callback:^(ARTTokenDetails *tokenDetails,
+                                                              NSError *error) {
+    ARTRealtime *realtime = [[ARTRealtime alloc] initWithToken:tokenDetails.token];
+    ARTRealtimeChannel *channel = [realtime.channels get:@"realtime-chat"];
+
+    [channel.presence subscribe:ARTPresenceEnter callback:^(ARTPresenceMessage *member) {
+        NSLog(@"%@ entered realtime-chat", member.clientId);
+    }];
+
+    [channel.presence enterClient:@"Bob" data:nil]; // prints 'Bob entered realtime-chat'
+    [channel.presence enterClient:@"Mary" data:nil]; // prints 'Mary entered realtime-chat'
+}];
+```
+
+```[swift]
+let rest = ARTRest(key: "{{API_KEY}}")
+// request a wildcard token
+rest.auth.requestToken(ARTTokenParams(clientId: "*"), withOptions: nil) { tokenDetails, error in
+    let realtime = ARTRealtime(token: tokenDetails!.token)
+    let channel = realtime.channels.get("realtime-chat")
+
+    channel.presence.subscribe(.enter) { member in
+        print("\(member.clientId) entered realtime-chat")
+    }
+
+    channel.presence.enterClient("Bob", data: nil) // prints 'Bob entered realtime-chat'
+    channel.presence.enterClient("Mary", data: nil) // prints 'Mary entered realtime-chat'
+}
+```
+
+h4(#rest-batch-publish). REST API: Batch presence
+
+It is common for the presence of multiple channels to be needed. If you wish to obtain the presence of multiple channels within a single operation, you can make use of the "REST batch API":/rest/batch.
+
+
+h3(#handling-failures). Handling transient channel/connection failures
+
+h4(#realtime-handling-failures). Realtime API
+
+Any time a channel is re-attached and the presence set is re-synced, e.g. after a short disconnection, the client will check whether any members it has entered into the presence set are there. If not, it will automatically re-enter them. This means that if a channel loses continuity (for example, because a client was disconnected from Ably for more than two minutes before reconnecting), then after the channel automatically re-attaches, any presence members it had previously entered will be restored.
+
+The exception is if you use the "recover":/realtime/connection#connection-state-recovery feature to resume a previous connection with a fresh client library instance (for example, to have continuity over a page refresh). In that case you will need to explicitly re-enter presence after you re-attach to the channel, due to the loss of client library internal state.
+
+h3(#unstable-connections). Presence with unstable connections
+
+h4(#realtime-unstable-connections). Realtime API
```

Comment:
There does not seem to be a corresponding REST section, so I think the h4 here might not be needed.